src/ngraph/op/fused/log_softmax.cpp

```diff
@@ -0,0 +1,51 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <numeric>
+
+#include "ngraph/op/fused/log_softmax.hpp"
+#include "ngraph/op/log.hpp"
+#include "ngraph/op/softmax.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::LogSoftmax::type_info;
+
+op::LogSoftmax::LogSoftmax(const Output<Node>& data, int64_t axis)
+    : FusedOp({data})
+    , m_axis(axis)
+{
+    constructor_validate_and_infer_types();
+}
+
+NodeVector op::LogSoftmax::decompose_op() const
+{
+    const auto data = input_value(0);
+    const auto data_shape = data.get_shape();
+
+    std::vector<size_t> axes(data_shape.size() - m_axis);
```

Comment:
what if `m_axis` is negative?

Reply:
According to arrangements I move handling negative axis from `onnx_importer` to `ngraph op`